modules/nlu/src/backend/prediction-handler.ts

```diff
@@ -0,0 +1,64 @@
+import * as sdk from 'botpress/sdk'
+
+type ModelGetter = (lang: string) => Promise<sdk.NLU.Model | undefined>
+
+export class PredictionHandler {
+  constructor(private getLatestModel: ModelGetter, private engine: sdk.NLU.Engine, private defaultLanguage: string) {}
+
+  async predict(textInput: string, includedContexts: string[]) {
+    const defaultResults = await this.engine.predict(textInput, includedContexts, this.defaultLanguage)
```

Comment:
do we always predict twice (if detected isn't default) even in NLU Wrapper ? 

Didn't we say that in wrapper we would run language detection prior to whole predict ? maybe i'm missing somthing